### PR TITLE
Refactoring/fairspc 79

### PR DESCRIPTION
**Pull request by tgreenwood**

### CODE REVIEW

File: projects/saturn/src/main/java/io/fairspace/saturn/rdf/transactions/SimpleTransactions.java [hunk 1]
```diff
@@ -4,10 +4,19 @@
...
 import org.apache.jena.system.Txn;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.stereotype.Component;
 
+@Component
+@ConditionalOnProperty(
+        name = "application.jena.bulkTransactions",
+        havingValue = "false",
+        matchIfMissing = true // initialize the Simple implementation absent
```
Comment (ewelinajozwik) [inline projects/saturn/src/main/java/io/fairspace/saturn/rdf/transactions/SimpleTransactions.java]:
```suggestion
        matchIfMissing = true // initialize the Simple implementation if absent
```

File: projects/saturn/src/test/java/io/fairspace/saturn/services/metadata/ReadableMetadataServiceTest.java [hunk 2]
```diff
@@ -41,15 +40,21 @@ public class ReadableMetadataServiceTest {
...
     private Transactions txn = new SimpleTransactions(createTxnMem());
     private MetadataService api;
-    private Model vocabulary = SYSTEM_VOCABULARY.union(createDefaultModel());
+    private final Model vocabulary;
+    private final Model systemVocabulary;
+
+    {
+        systemVocabulary = loadModel("system-vocabulary.ttl");
+        vocabulary = systemVocabulary.union(createDefaultModel());
+    }
```
Comment (ewelinajozwik) [inline projects/saturn/src/test/java/io/fairspace/saturn/services/metadata/ReadableMetadataServiceTest.java]: Why like this, not as variables?

Comment (tgreenwood) [reply projects/saturn/src/test/java/io/fairspace/saturn/services/metadata/ReadableMetadataServiceTest.java]: probably I went crazy when I was writing this :) fixed

File: projects/saturn/src/main/java/io/fairspace/saturn/rdf/transactions/SimpleTransactions.java (hunk 1 above)
Comment (tgreenwood) [reply projects/saturn/src/main/java/io/fairspace/saturn/rdf/transactions/SimpleTransactions.java]: agree, made bulk as a default one

File: projects/saturn/src/main/java/io/fairspace/saturn/webdav/resources/DirectoryResource.java [hunk 3]
```diff
@@ -358,6 +374,14 @@ private void uploadMetadata(FileItem file) throws BadRequestException, ConflictE
             throw new BadRequestException("Error parsing file " + file.getName(), e);
         }
 
+        // TODO: I would refactor all resources to be anemic rather than rich.
```
Comment (ewelinajozwik) [inline projects/saturn/src/main/java/io/fairspace/saturn/webdav/resources/DirectoryResource.java]: TODO

**Pull request merged**